R/berekenVerschilscores.R

```diff
@@ -55,16 +59,31 @@ berekenVerschilscores <-
         Waarde = (.data$WaardeMin + .data$WaardeMax) / 2,
         # negatieve indicatoren * -1
         Teken = ifelse(.data$Operator %in% c("<=", "<"), -1, 1),
+        Teken =
+          ifelse(
+            .data$Ref <= 0 & .data$Operator == "=", -.data$Teken, .data$Teken
+          ),
+        Ref = abs(.data$Ref),
         BereikGunstig = ifelse(.data$Operator %in% c("<=", "<"),
                                .data$Ref,
                                .data$TheoretischMaximum - .data$Ref),
         BereikOngunstig = .data$TheoretischMaximum - .data$BereikGunstig,
+        BereikOngunstig =
+          ifelse(
+            is.na(.data$BereikOngunstig) & !.data$Operator %in% c("<=", "<"),
+            .data$Ref,
+            .data$BereikOngunstig
+          ),
         Verschil = (.data$Waarde - .data$Ref) * .data$Teken,
         Verschilscore = case_when(
           is.na(.data$Verschil) ~ ifelse(.data$Waarde == .data$Ref, 1, -1),
           #gewijzigd naar >= 0 (bereik gunstig is inclusief de grenswaarde zelf)
-          .data$Verschil >= 0 ~ .data$Verschil / .data$BereikGunstig,
-          .data$Verschil < 0 ~ .data$Verschil / .data$BereikOngunstig
+          .data$Verschil >= 0 & .data$BereikGunstig != 0 ~
+            .data$Verschil / .data$BereikGunstig,
+          .data$Verschil >= 0 & .data$BereikGunstig == 0 ~ 1,
+          .data$Verschil < 0 & .data$BereikOngunstig != 0 ~
+            .data$Verschil / .data$BereikOngunstig,
+          .data$Verschil < 0 & .data$BereikOngunstig == 0 ~ -1
```

Comment:
lijn 86 is wellicht overbodig (maar kan ook geen kwaad): verschil kan niet < 0 zijn als bereik ongunstig == 0.